### PR TITLE
refactor: removed unused code and unified tx signing logic

### PR DESCRIPTION
**Pull request by ws4charlie**
# Description

The PR is a minimum refactor to prepare for EIP-1559 outbound transaction

- removed unused method `SignWhitelistTx`
- unified different tx signing (onReceive, on Revert, erc20 withdraw, gas token withdraw) to use a common method `Sign`.
- unified the tx verification logic in unit tests; added missing unit tests for `SignCancelTx` and `TestSigner_SignMigrateTssFundsCmd`
- fix 'amount' issue in `SignMigrateTssFundsCmd`

Closes: <PD-XXXX>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: zetaclient/chains/evm/signer/signer.go [hunk 1]
```diff
@@ -207,12 +213,13 @@ func (signer *Signer) SignOutbound(txData *OutboundData) (*ethtypes.Transaction,
 
 	tx, _, _, err := signer.Sign(data,
 		signer.zetaConnectorAddress,
+		big.NewInt(0),
```
Comment (kingpinXD) [inline zetaclient/chains/evm/signer/signer.go:216]: Consider defining the zero amount somewhere as a constant.

**Review by kingpinXD — APPROVED** [review]

File: zetaclient/chains/evm/signer/signer.go [hunk 2]
```diff
@@ -39,7 +39,10 @@ import (
 	"github.com/zeta-chain/zetacore/zetaclient/zetacore"
 )
 
-var _ interfaces.ChainSigner = &Signer{}
+var (
+	zeroValue                        = int64(0)
```
Comment (lumtis) [inline zetaclient/chains/evm/signer/signer.go:43]: Seems like to be only used as big.Int, why not directly defining it as big.Int?

Also need comment why the variable is necessary

File: zetaclient/chains/evm/signer/signer.go [hunk 3]
```diff
@@ -241,74 +246,57 @@ func (signer *Signer) SignRevertTx(txData *OutboundData) (*ethtypes.Transaction,
...
+		signer.tssSigner.EVMAddress(),
+		big.NewInt(zeroValue), // zero out the amount to cancel the tx
+		evm.EthTransferGasLimit,
+		txData.gasPrice,
+		txData.nonce,
+		txData.height,
+	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("SignCancelTx error: %w", err)
```
Comment (lumtis) [inline zetaclient/chains/evm/signer/signer.go:278]: Not covered by test

File: zetaclient/chains/evm/signer/signer.go [hunk 4]
```diff
@@ -241,74 +246,57 @@ func (signer *Signer) SignRevertTx(txData *OutboundData) (*ethtypes.Transaction,
...
+		txData.to,
+		txData.amount,
+		evm.EthTransferGasLimit,
+		txData.gasPrice,
+		txData.nonce,
+		txData.height,
+	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("SignWithdrawTx error: %w", err)
```
Comment (lumtis) [inline zetaclient/chains/evm/signer/signer.go:296]: Not covered by test, sme for other sign error below

**Review by lumtis — COMMENTED** [review]
This looks legit to me. Some few comments.

Wondering if we could think about a further refactoring with a declarative approach?
We have a single sign function and an array containing the different values depending on the tx to sign.
Then to support a new tx that can be signed, you only need to add an entry in this array

File: zetaclient/chains/evm/signer/signer.go (hunk 3 above)
Comment (ws4charlie) [reply zetaclient/chains/evm/signer/signer.go:280]: added more tests in https://github.com/zeta-chain/node/pull/2306/commits/3db4bef06ed6dee6a142efe5152d1704d9084793

File: zetaclient/chains/evm/signer/signer.go (hunk 4 above)
Comment (ws4charlie) [reply zetaclient/chains/evm/signer/signer.go:298]: same commit https://github.com/zeta-chain/node/pull/2306/commits/3db4bef06ed6dee6a142efe5152d1704d9084793

**Comment by ws4charlie** [comment]
> This looks legit to me. Some few comments.
> 
> Wondering if we could think about a further refactoring with a declarative approach? We have a single sign function and an array containing the different values depending on the tx to sign. Then to support a new tx that can be signed, you only need to add an entry in this array

I think we can create a issue to track this: https://github.com/zeta-chain/node/issues/2318

**Review by lumtis — APPROVED** [review]

**Pull request merged**